packages/amplify-category-api/commands/api/add-graphql-datasource.js

```diff
@@ -205,3 +207,42 @@ async function getAwsClient(context, action) {
   const provider = require(providerPlugins[providerName]);
   return await provider.getConfiguredAWSClient(context, 'aurora-serverless', action);
 }
+
+function readSchema(projectDirectory, rdsSchema) {
+  const schemaFilePath = path.join(projectDirectory, 'schema.graphql');
+  const schemaDirectoryPath = path.join(projectDirectory, 'schema');
+  const schemaDirectoryExists = fs.existsSync(schemaDirectoryPath);
+  // convert the object schema to dump as file
+  const rdsSchemaDoc = mergeTypes([{}, rdsSchema], { all: true });
+
+  let schema;
+  if (schemaDirectoryExists) {
+    schema = readSchemaDocuments(schemaDirectoryPath).join('\n');
+    const rdsSchemaFilePath = path.join(schemaDirectoryPath, 'rds.graphql');
+    fs.writeFileSync(rdsSchemaFilePath, rdsSchemaDoc, 'utf8');
```

Comment:
Why is the file being written here when this function is call `readSchema`?

Reply:
fixed